sled-agent/src/sled_agent.rs

```diff
@@ -401,6 +411,19 @@ impl SledAgent {
             rack_network_config.clone(),
         )?;
 
+        // Spawn a task to periodically cleanup old zone bundles.
+        let (zone_bundle_tx, rx) =
+            mpsc::channel(zone_bundle::NUM_OUTSTANDING_CLEANUP_REQUESTS);
+        let cleanup_log = log.new(o!("component" => "ZoneBundleCleanup"));
+        let cleanup_context = zone_bundle::CleanupContext::with_defaults(
+            &storage.resources().all_zone_bundle_directories().await,
+        )?;
```

Comment:
So, the zone bundles are currently on the M.2s. We talked about this a while ago, but the goal was just to sidestep a bunch of issues with storing them on the U.2s, one of which is exactly this. Imagine you unplug a U.2; you do something like pick a new U.2 on which to store zone bundles; and then the U.2 is reinserted. The cleanup task needs to know about _all_ of those locations, since it may need to remove bundles from a dataset that's no longer used to _store_ bundles. It just gets wonky, and I think we can avoid a lot of that by storing them on the M.2s at this point.

I'm happy to pass a reference to the storage manager or resources right now though. Even if we still keep them on the M.2s, it'll be easier to generalize in the future.

Reply:
The new `ZoneBundler` object takes a reference to the `StorageResources` on creation now, in a70596cc. It ensures that all the directories it needs exist whenever they're accessed. I can't be sure how that'll play with hot-swapped U.2s in the future, but this should be sufficient while bundles live on the M.2s.